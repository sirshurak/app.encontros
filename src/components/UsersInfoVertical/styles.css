#users-info-vertical {
    display: flex;
    flex-direction: column;
}

#users-info-vertical h2 {
    text-align: left;
}

#users-info-vertical button.link {
    padding: 10px 0px;
}