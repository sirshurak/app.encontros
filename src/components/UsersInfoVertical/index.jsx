import React from "react";
import { Container } from "react-bootstrap";
import UserInfo from "../UserInfo";
import UserInfoClick from "../UserInfoClick";

import "./styles.css";

function UsersInfoVertical({ users, title, ageTitle }) {
  return (
    <section id="users-info-vertical">
      <Container>
        <h2>{title}</h2>
        {users?.length &&
          users.map((user, key) => (
            <UserInfoClick user={user} key={key}>
              <UserInfo {...user} variant="vertical" ageTitle={ageTitle} />
            </UserInfoClick>
          ))}
      </Container>
    </section>
  );
}

export default UsersInfoVertical;
