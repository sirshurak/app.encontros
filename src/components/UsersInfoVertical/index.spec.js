import React from 'react';
import { shallow, mount } from 'enzyme';
import UsersInfoVertical from './index';
import AppContext from '../../contexts/AppContext';

const users = [{
    photo: 'https://picsum.photos/200/?image=26',
    username: 'username',
    city: 'Test City',
    gender: 'Man',
    sexualOrientation: 'Heterosexual',
    age: 27
}];

describe('UsersInfoVertical component', () => {
    it('should match shallow snapshop', () => {
        const component = shallow(<UsersInfoVertical title="Title" ageTitle="anos" users={users}/>)
        expect(component).toMatchSnapshot();
    });

    it('should mount slider items', () => {
        const changePage = jest.fn();
        const component = mount(
            <AppContext.Provider value={{ changePage }}>
                <UsersInfoVertical title="Title" ageTitle="anos" users={users}/>
            </AppContext.Provider>
        );

        expect(component.find('h3').text()).toBe('username');
    });
});