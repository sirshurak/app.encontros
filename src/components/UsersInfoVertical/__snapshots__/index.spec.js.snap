// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersInfoVertical component should match shallow snapshop 1`] = `
<section
  id="users-info-vertical"
>
  <Container
    fluid={false}
  >
    <h2>
      Title
    </h2>
    <UserInfoClick
      key="0"
      user={
        Object {
          "age": 27,
          "city": "Test City",
          "gender": "Man",
          "photo": "https://picsum.photos/200/?image=26",
          "sexualOrientation": "Heterosexual",
          "username": "username",
        }
      }
    >
      <UserInfo
        age={27}
        ageTitle="anos"
        city="Test City"
        gender="Man"
        photo="https://picsum.photos/200/?image=26"
        sexualOrientation="Heterosexual"
        username="username"
        variant="vertical"
      />
    </UserInfoClick>
  </Container>
</section>
`;
